docs: Deploying a Static Site npm command instructions inconsistent with template package.json

### Describe the bug

Going through the documentation to [Deploy a Static Site](https://vitejs.dev/guide/static-deploy.html) the npm commands listed to see a preview of the built project are inconsistent with what the template `package.json` that's placed in the root directory of the project. 

I went through the the templates for vue, react, preact, and vanilla and their respective Typescript counterparts. The templates all have the scripts command `serve` instead of the documented `preview` command. 

Potential options:
1. Update the documentation to reflect the template `package.json` file commands `serve` to run `vite preview`.
2. Update the templates to include both `serve` and `preview` to run `vite preview`. Either issuing a deprecation warning when running `npm run serve` or making note of the breaking change in the next 2.x release.

[Original discussion from Discord](https://discord.com/channels/804011606160703521/804439875226173480/876101237748146227)

### Reproduction

Creating a new project using the documentation:
```sh
npm init vite@latest --template <vue | vue-ts | vanilla | vanilla-ts | react | react-ts | preact | preact-ts> test-project
cd ./test-project
npm install
npm run build
npm run preview
```


### System Info

```shell
System:
    OS: Linux 5.13 Fedora 34 (Workstation Edition) 34 (Workstation Edition)
    CPU: (20) x64 Intel(R) Core(TM) i9-10900KF CPU @ 3.70GHz
    Memory: 7.23 GB / 31.26 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.5 - ~/.nvm/versions/node/v14.17.5/bin/node
    Yarn: 1.22.10 - /usr/bin/yarn
    npm: 6.14.14 - ~/.nvm/versions/node/v14.17.5/bin/npm
  Browsers:
    Firefox: 90.0.2
  npmPackages:
    @vitejs/plugin-vue: ^1.3.0 => 1.4.0
    vite: ^2.4.4 => 2.4.4
```


### Used Package Manager

npm

### Logs

```shell
npm run preview
npm ERR! missing script: preview

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/theoriginalstove/.npm/_logs/2021-08-15T05_57_12_414Z-debug.log
```


### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/vue-next instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/docs/guide/static-deploy.md
+++ b/docs/guide/static-deploy.md
@@ -40,7 +40,7 @@ $ npm run build
 $ npm run serve
 ```
 
-The `preview` command will boot up local static web server that serves the files from `dist` at http://localhost:5000. It's an easy way to check if the production build looks OK in your local environment.
+The `vite preview` command will boot up local static web server that serves the files from `dist` at http://localhost:5000. It's an easy way to check if the production build looks OK in your local environment.
 
 You may configure the port of the server py passing `--port` flag as an argument.
 

--- a/docs/guide/static-deploy.md
+++ b/docs/guide/static-deploy.md
@@ -10,12 +10,12 @@ The following guides are based on some shared assumptions:
 {
   "scripts": {
     "build": "vite build",
-    "preview": "vite preview"
+    "serve": "vite preview"
   }
 }
 ```
 
-It is important to note that `vite preview` is intended for previewing the build locally and not meant as a production server.
+It is important to note that `vite serve` is intended for previewing the build locally and not meant as a production server.
 
 ::: tip NOTE
 These guides provide instructions for performing a static deployment of your Vite site. Vite also has experimental support for Server Side Rendering. SSR refers to front-end frameworks that support running the same application in Node.js, pre-rendering it to HTML, and finally hydrating it on the client. Check out the [SSR Guide](./ssr) to learn about this feature. On the other hand, if you are looking for integration with traditional server-side frameworks, check out the [Backend Integration guide](./backend-integration) instead.
@@ -33,11 +33,11 @@ By default, the build output will be placed at `dist`. You may deploy this `dist
 
 ### Testing The App Locally
 
-Once you've built the app, you may test it locally by running `npm run preview` command.
+Once you've built the app, you may test it locally by running `npm run serve` command.
 
 ```bash
 $ npm run build
-$ npm run preview
+$ npm run serve 
 ```
 
 The `preview` command will boot up local static web server that serves the files from `dist` at http://localhost:5000. It's an easy way to check if the production build looks OK in your local environment.
@@ -47,7 +47,7 @@ You may configure the port of the server py passing `--port` flag as an argument
 ```json
 {
   "scripts": {
-    "preview": "vite preview --port 8080"
+    "serve": "vite preview --port 8080"
   }
 }
 ```

--- a/docs/guide/static-deploy.md
+++ b/docs/guide/static-deploy.md
@@ -54,6 +54,10 @@ You may configure the port of the server py passing `--port` flag as an argument
 
 Now the `preview` method will launch the server at http://localhost:8080.
 
+::: tip NOTE
+If you are using `npm` as your package manager, you may run into issues with changing the `serve` command to `preview` due to npm's usage of `pre` and `post` commands. For more info checkout the [npm docs](https://docs.npmjs.com/cli/v7/using-npm/scripts#pre--post-scripts) regarding Pre & Post scripts.
+:::
+
 ## GitHub Pages
 
 1. Set the correct `base` in `vite.config.js`.

--- a/docs/guide/static-deploy.md
+++ b/docs/guide/static-deploy.md
@@ -37,7 +37,7 @@ Once you've built the app, you may test it locally by running `npm run serve` co
 
 ```bash
 $ npm run build
-$ npm run serve 
+$ npm run serve
 ```
 
 The `preview` command will boot up local static web server that serves the files from `dist` at http://localhost:5000. It's an easy way to check if the production build looks OK in your local environment.

--- a/docs/guide/static-deploy.md
+++ b/docs/guide/static-deploy.md
@@ -15,7 +15,7 @@ The following guides are based on some shared assumptions:
 }
 ```
 
-It is important to note that `vite serve` is intended for previewing the build locally and not meant as a production server.
+It is important to note that `vite preview` is intended for previewing the build locally and not meant as a production server.
 
 ::: tip NOTE
 These guides provide instructions for performing a static deployment of your Vite site. Vite also has experimental support for Server Side Rendering. SSR refers to front-end frameworks that support running the same application in Node.js, pre-rendering it to HTML, and finally hydrating it on the client. Check out the [SSR Guide](./ssr) to learn about this feature. On the other hand, if you are looking for integration with traditional server-side frameworks, check out the [Backend Integration guide](./backend-integration) instead.

--- a/docs/guide/static-deploy.md
+++ b/docs/guide/static-deploy.md
@@ -55,7 +55,7 @@ You may configure the port of the server py passing `--port` flag as an argument
 Now the `preview` method will launch the server at http://localhost:8080.
 
 ::: tip NOTE
-If you are using `npm` as your package manager, you may run into issues with changing the `serve` command to `preview` due to npm's usage of `pre` and `post` commands. For more info checkout the [npm docs](https://docs.npmjs.com/cli/v7/using-npm/scripts#pre--post-scripts) regarding Pre & Post scripts.
+If you change the script name from `serve` to `preview`, you may run into issues with some package managers due to the way they handle [Pre & Post scripts](https://docs.npmjs.com/cli/v7/using-npm/scripts#pre--post-scripts).
 :::
 
 ## GitHub Pages

